### PR TITLE
Added configuration Files for Jest and EsLint

### LINKED ISSUE #16: Es-lint
**Describe your feature**

Set up es-lint to use tabs and not space for indent.
Enforce the javascript standards for best coverage.

**You can add additional information like, screenshots here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by linuxuser07**
# Description

Added the Jest configuration and also the Eslint Configuration.

Fixes #16 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        "commonjs": true,
+        "es6": true,
+        "node": true
+    },
+    "extends": "eslint:recommended",
+    "parserOptions": {
+        "ecmaVersion": 2016
+    },
+    "rules": {
+        "indent": [
```
Comment (dashcraft) [inline .eslintrc.js:13]: We need to remove the indent rule, we want to ignore file endings at the moment, we work in windows and linux, git will control file endings for us.

File: package.json [hunk 2]
```diff
@@ -20,24 +20,30 @@
...
+    "modulePathIgnorePatterns": [
+      "coverage"
     ]
   },
   "scripts": {
     "test": "jest --verbose",
     "test:coverage": "npm test -- --coverage",
-    "test:badges": "npm run test:coverage  && jest-coverage-badges"
+    "test:badges": "npm run test:coverage  && jest-coverage-badges",
+    "lint": "eslint"
```
Comment (dashcraft) [inline package.json]: this needs to be "lint":"eslint ."

to point eslint to start int he current directory

**Review by dashcraft — CHANGES_REQUESTED** [review]
Check out the comments below and then resubmit the corrected pr.

File: .eslintrc.js (hunk 1 above)
Comment (linuxuser07) [reply .eslintrc.js:13]: I do have a question, in the Issue #16  it mentions to use tabs for indent and not spaces.

Comment (dashcraft) [reply .eslintrc.js:13]: I meant the lineEndings rule, i apologize, not the indent rule.

**Review by dashcraft — APPROVED** [review]

**Pull request merged**